CenterForOpenScience/osf.io
# Remove the is_claimed field from the user model [PLAT-1055]

## Purpose
There was an old `#TODO` in `user.py` that mentioned that the `is_claimed` field might not be in use, as it was pretty much mirroring `is_registered`. While most users had both of these fields as the same, there were quite a few where they were different. Turns out this could probably be because [the disable user view](https://github.com/CenterForOpenScience/osf.io/blob/develop/admin/users/views.py#L60) sets `is_registered = False` and does not change `is_claimed`. 

With some help from @mfraezz, confirmed indeed that on a prod dump
`OSFUser.objects.exclude(is_claimed=F('is_registered')).count()`
is the same as
`OSFUser.objects.exclude(is_claimed=F('is_registered')).filter(date_disabled__isnull=False).count()`
and those users with the descrepency did indeed all have `is_claimed` still set to `True`.

Because of this, pretty sure we can safely remove the field and change the usages of `is_claimed` to just use `is_registered`.

## Changes

- Remove the `is_claimed` field from the model, add migration
- Change old usages of `is_claimed` to `is_registered`

## QA Notes
This is mostly not a user facing change. The one user-facing spot seems to be determining if a disabled user's contributor name is a link or not on a project's contributor list, so that would be a good place to spot check!


## Documentation

None needed

## Side Effects

None anticipated

## Ticket

https://openscience.atlassian.net/browse/PLAT-1055
